pkg/serializer/compression/zlib.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build zlib && !zstd
+
+// Package compression provides a set of functions for compressing with zlib / zstd
+package compression
+
+import (
+	"bytes"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	compression "github.com/DataDog/datadog-agent/pkg/serializer/compression/internal"
+	"github.com/DataDog/datadog-agent/pkg/serializer/compression/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// NewCompressorStrategy returns a new Compressor based on serializer_compressor_kind
+// This function is called only when the zlib build tag is included
+func NewCompressorStrategy(cfg config.Component) utils.Compressor {
+	kind := cfg.GetString("serializer_compressor_kind")
+	switch kind {
+	case utils.ZlibKind:
+		return compression.NewZlibStrategy()
+	case utils.ZstdKind:
+		log.Warn("zstd build tag not included. using zlib")
+		return compression.NewZlibStrategy()
+	case utils.NoneKind:
+		return compression.NewNoopStrategy()
```

Comment:
Should we have a warning here as well?
```suggestion
        log.Warn("No serializer_compressor_kind detected. use zlib or zstd")
		return compression.NewNoopStrategy()
```